EEG101/src/components/NavBar.js

```diff
@@ -31,8 +39,151 @@ function mapDispatchToProps(dispatch) {
 class NavBar extends Component {
   constructor(props) {
     super(props);
+    this.predictSubscription = {};
+	
+    this.state = {
+      batteryLevel: '',
+    };
+  }
+  
+  componentDidMount () {
+    TimerMixin.setInterval(() => { 
+	  this.props.setRefresh(!this.props.refresh); 
+	  this.startBatteryReading();
+      this.stopBatteryReading();
+	}, 1000);
+  }
+  
+   componentWillUnmount() {
+    this.props.setNoMore(this.props.noMore = false);
+  }
+
+  startBatteryReading() {
+    if (this.props.connectionStatus === config.connectionStatus.CONNECTED) {
+      if (this.props.noMore === false) {
+        Battery.startReading();
+        const batteryListener = new NativeEventEmitter(
+          NativeModules.Battery
+        );
+        this.predictSubscription = batteryListener.addListener(
+          "BATTERY", battery => {
+            this.state.batteryLevel = battery;
+          }
+        );
+        this.props.setNoMore(this.props.noMore = true);
```

Comment:
If `stopReading` is getting called immediately after startReading, is it necessary to have this separate noMore variable?

Reply:
It is being called only one time after muse is disconnected. Explanation in the comment below.